uplink/config.go

```diff
@@ -40,10 +43,58 @@ type RSConfig struct {
 // EncryptionConfig is a configuration struct that keeps details about
 // encrypting segments
 type EncryptionConfig struct {
-	Key       string      `help:"root key for encrypting the data"`
-	BlockSize memory.Size `help:"size (in bytes) of encrypted blocks" default:"1KiB"`
-	DataType  int         `help:"Type of encryption to use for content and metadata (1=AES-GCM, 2=SecretBox)" default:"1"`
-	PathType  int         `help:"Type of encryption to use for paths (0=Unencrypted, 1=AES-GCM, 2=SecretBox)" default:"1"`
+	// TODO: WIP#if/v3-1541 read TODO in Key method why the following field cannot
+	// be added
+	// RootKey     *storj.Key  // The root key for encrypting the data and read it from KeyFilepath, see Key method.
+	KeyFilepath string      `help:"the path to the file which contains the root key for encrypting the data"`
+	BlockSize   memory.Size `help:"size (in bytes) of encrypted blocks" default:"1KiB"`
+	DataType    int         `help:"Type of encryption to use for content and metadata (1=AES-GCM, 2=SecretBox)" default:"1"`
+	PathType    int         `help:"Type of encryption to use for paths (0=Unencrypted, 1=AES-GCM, 2=SecretBox)" default:"1"`
+}
+
+// Key returns the root key for encrypting the data which is stored in the path
+// indicated by KeyFilepath field.
+//
+// It returns an error if:
+//
+// * The file doesn't exist.
+// * The file doesn't have exactly 0400 permissions.
+// * There is an I/O error.
+//
+// TODO: WIP#if/v3-1541 refactor cache the key after the firs read. The problem
+// is with pkg/cfgstruct which is inflexible on accepting unexported keys nor
+// several different types as a type of storj.Key, [32]byte, etc.
+func (encCfg *EncryptionConfig) Key() (storj.Key, error) {
+	if encCfg.KeyFilepath == "" {
+		return storj.Key{}, nil
+	}
+
+	f, err := os.Open(encCfg.KeyFilepath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return storj.Key{}, Error.Wrap(fmt.Errorf("not found key file %q", encCfg.KeyFilepath))
+		}
+
+		return storj.Key{}, errs.Wrap(err)
```

Comment:
`Erros.Wrap`?

Reply:
I doubted it if it should use for errors not directly related to `Uplink configuration`.

Can you confirm that all the errors on this package should be of such class?
My understanding from this comment is yes, because it's mostly a way to prefix the error message, but I'm not entirely sure if that's correct.

Reply:
ok, I'm also not fully sure if this should be wrapped with uplink error

Reply:
Ok, then let's ask to others

Do @kaloyan-raev @mobyvb know more how this is applied?

Reply:
I think we always prefer using `Error.Wrap(err)`. Prefixing the error makes it easier to spot the context in which the error happened.

Also, I think that failing to open the file with the encryption key is an error that is very well related to the uplink configuration.